main.go

```diff
@@ -92,3 +98,66 @@ func retrieveComments(segment string) *string {
 	}
 	return completion
 }
+
+func evaluateComments(pr *github.PullRequest, client github_client.GithubRepoClient) error {
+	// List review comments on a pull request
+	comments, err := client.GetPrComments(pr)
+	if err != nil {
+		log.Fatalf("Failed to get comments: %v", err)
+	}
+
+	for _, commentDetails := range comments {
+		comment := commentDetails.GetBody()
+		diffHunk := commentDetails.GetDiffHunk()
+
+		openaiClient := openai.OpenAiClient{}
+		openaiClient.Init()
+		conversation := []openai_api.ChatCompletionMessage{
+			{
+				Role:    openai_api.ChatMessageRoleSystem,
+				Content: `You are an expert code review quality evaluator.`,
+			},
+			{
+				Role:    openai_api.ChatMessageRoleUser,
+				Content: fmt.Sprintf("Please summarize what this code is doing:\n%s", diffHunk),
+			},
+		}
+		completion, err := openaiClient.RequestCompletion(conversation)
+		if err != nil {
+			return err
+		}
+		conversation = append(conversation, openai_api.ChatCompletionMessage{
+			Role:    openai_api.ChatMessageRoleAssistant,
+			Content: completion,
+		}, openai_api.ChatCompletionMessage{
+			Role:    openai_api.ChatMessageRoleUser,
+			Content: fmt.Sprintf("Please rate the quality of this code review comment on a scale of 1-5:\n%s\nOnly respond with the number. ALL COMMENTS ABOUT IMPORTS GET A 1.", comment),
+		})
+		final, err := openaiClient.RequestCompletion(conversation)
+		if err != nil {
+			return err
+		}
+
+		// convert "final" first char to int
+		score, err := strconv.Atoi(final[0:1])
+		if err != nil {
+			return err
+		}
+		if score < 3 {
```

Comment:
Bug: The string conversion should use a more robust method to ensure that only valid numbers are input.

Reply:
fixed.